src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -72,17 +63,24 @@ public boolean isOwner(NsUser loginUser) {
         return writer.equals(loginUser);
     }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
-
     public boolean isDeleted() {
         return deleted;
     }
 
-    public List<Answer> getAnswers() {
-        return answers;
+    public List<DeleteHistory> delete(NsUser loginUser) throws CannotDeleteException {
+        if (!isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
+
+        if (!answers.checkAllOwner(loginUser)) {
```

Comment:
```suggestion
        if (!answers.isAllOwner(loginUser)) {
```
일반적인 `isXXX` 같은 표현도 좋을 것 같아요
저는 `validateXXX`, `checkXXX` 네이밍은 내부에서 예외를 던질 때 주로 사용하고 있습니다.

개인적인 의견이니 참고만 해주세요~